apps/prairielearn/src/ee/auth/lti13/lti13auth.ts

```diff
@@ -0,0 +1,268 @@
+import { Router, type Request, Response, NextFunction } from 'express';
+import asyncHandler = require('express-async-handler');
+import { Issuer, Strategy, type StrategyVerifyCallbackReq, IdTokenClaims } from 'openid-client';
+import passport = require('passport');
+import { z } from 'zod';
+import { get as _get, some as _some } from 'lodash';
+
+import { loadSqlEquiv, queryAsync } from '@prairielearn/postgres';
+import error = require('@prairielearn/error');
+import * as authnLib from '../../../lib/authn';
+import { selectLti13Instance } from '../../models/lti13Instance';
+import { get as cacheGet, set as cacheSet } from '../../../lib/cache';
+import { getInstitutionAuthenticationProviders } from '../../lib/institution';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router({ mergeParams: true });
+
+const lti13_issuers = {};
+
+// Middleware to cache passport setup
```

Comment:
Rather than this, here's an idea: in `launchFlow` and `authenticate`, create a local instance of passport:

```ts
const daveLocalPassport = new passport.Passport();
daveLocalPassport.use('openid', new Strategy());
daveLocalPassport.authenticate('openid', {})(req, res, next);
```

This way, we don't have to worry about caching or polluting the "global" passport object.

Reply:
Done in ed977ea